### ISSUE TITLE
liveness e2e test is flaky

### ISSUE THREAD

**Issue by jlowdermilk** [GoogleCloudPlatform/kubernetes#4501]
63/78% passing for last 100 runs on [jenkins](http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce/lastCompletedBuild/testReport/(root)/Kubernetes%20e2e%20Suite/). Test itself is likely racy.

**Comment by zmerlynn** [comment]
I wasn't sure if this was the test or not: #4406.

**Comment by a-robinson** [comment]
Should this be part of a rewrite in go?

**Comment by a-robinson** [comment]
Seems like work on this should definitely wait for the in-flight PR on #4253

**Comment by mbforbes** [comment]
I was looking at this today.

Here's a doc of my quest:
https://docs.google.com/a/google.com/document/d/1UBNp-0_3LV9-3LmsWbRGU7FTyuKCwS9IqiCS4Xqk5LQ/pub

For Googlers, you should be able to directly comment on the doc:
https://docs.google.com/document/d/1UBNp-0_3LV9-3LmsWbRGU7FTyuKCwS9IqiCS4Xqk5LQ/edit#

tl;dr: I _think_ this is because our heavy stock pods (e.g. elasticsearch) are taking so long to come up, and we aren't waiting for them, that e2e tests that schedule pods can fail. A fix is to have the e2e test driver wait until all of a clusters' pods are "running" before starting to run tests. I'll be working on fully verifying and then fixing this.

**Comment by roberthbailey** [comment]
That sounds similar to what @alex-mohr raised in #4566.

**Comment by mbforbes** [comment]
Just as an update, doing this in `hack/e2e.go` but requiring utils from `e2e` package which breaks because of flags, refactoring is a mess because of other helpers using cross-file globals for test settings. So, this is "done" but requires more pluming to be PR-able. WIP...

**Comment by zmerlynn** [comment]
Can't you do this entirely in shell, in hack/e2e-internal/e2e-status.sh?

**Comment by zmerlynn** [comment]
Closed via #5135
